lib/msf/core/module.rb

```diff
@@ -122,6 +123,9 @@ def initialize(info = {})
     self.platform = PlatformList.transform(module_info['Platform'])
     self.references = Rex::Transformer.transform(module_info['References'], Array, [ SiteReference, Reference ], 'Ref')
 
+    # Create and initialize the notes container for this module
+    self.notes = Msf::NoteContainer.new(module_info['Notes'])
```

Comment:
Is there a reason we can't keep the original hash here? Sure, the surrounding code doesn't, but the surrounding code is often too complicated for it's own good.

Everywhere we access or use these values it seems we have to bend over backwards to make this object representation work. If we kept a normal `Hash`, the `dump_aka` check above would be `if mod.notes['AKA'].present?`, for example, which I feel is much cleaner.

Also, I think this sort of thing would be a better fit in `module_info.rb`, where we could add:
```ruby
def notes
  module_info['Notes']
end
```
instead of having the special `attr_accessor`.

Reply:
I have no issues moving the initialization into `module.info.rb`. However in leaving this as a plain hash, we'd lose the typing defined in `note.rb`. Is there any merit in keeping this typing, or will hash lookups be sufficient for future use cases?

Looping in @mkienow-r7 , since there's more discussion about the `Note` and `NoteContainer` type below

Reply:
It doesn't seem the value of these objects are being realized in any specific way in this PR. Did you have something greater planned for them, or are they simply a data object for the key-value pair?

Reply:
Reviving this conversation from the dead:
There's nothing that makes use of that object type right now. I was over-engineering in the event we ever do anything fancier with presentation and want a cleaner way to access these attributes. I'm all for leaving everything as a hash and crossing that bridge when we come to it. Seems like this implementation could get messy if we ever have data more complex than an array.

Reply:
I think it's fine to say that this is for arrays of strings only, and if anyone in the future needs something more complicated they can build a type validated metadata object themselves.